### PR TITLE
Reset password email tests

### PR DESCRIPTION
**Pull request by allait**
This adds step definitions for flows that include sending an email to the user.

By making notification references predictable we can use Notify API to retrieve the email by reconstructing the reference that the application attached to it in the functional test code.

This allows us to get the email body directly from Notify.

The benefit of this approach is that we can use test Notify API keys and avoid sending actual emails to fake email addresses. This is also a much simpler setup since it doesn't require setting up an email address to receive emails or adding delays to the tests to account for email delivery.

### CODE REVIEW

File: features/step_definitions/emails.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  end
+end
+
+Then /^I receive a '([a-z-]+)' email for #{MAYBE_VAR}/ do |message_type, email_address|
+  messages = DMNotify.get_email(message_type, email_address)
+  @email_text = messages.collection[0].body
+end
+
+Then /^I click the link in that email$/ do
+  page.visit(URI.extract(@email_text).select { |i| i.start_with?('http') } [0])
```
Comment (samuelhwilliams) [inline features/step_definitions/emails.rb]: Is it not going to always start with 'https'?

File: features/user/reset_password.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+@reset-password @notify
+Feature: Reset user password
+
+Scenario: Request password reset
+  Given I have a buyer user
+  And I am on the /user/reset-password page
+  When I enter that user.emailAddress in the 'Email address' field
+  And I click 'Send reset email' button
+  Then I see a success banner message containing 'send a link to reset the password'
```
Comment (samuelhwilliams) [inline features/user/reset_password.feature]: Hmm, it feels like the content might have been better as `If the email address you've entered belongs to a Digital Marketplace account, we've sent a link to reset the password.` ... But I guess that's a different matter.

File: features/step_definitions/emails.rb (hunk 1 above)
Comment (allait) [reply features/step_definitions/emails.rb]: Not when run locally

Comment (samuelhwilliams) [reply features/step_definitions/emails.rb]: Ah ok 👍

**Pull request merged**